src/gui/qgsgui.h

```diff
@@ -41,10 +41,22 @@ class QgsWindowManagerInterface;
  * related to GUI classes.
  * \since QGIS 3.0
  */
-class GUI_EXPORT QgsGui
+class GUI_EXPORT QgsGui : public QObject
 {
+    Q_OBJECT
```

Comment:
Is using `Q_GADGET` instead of `Q_OBJECT` possible here?

Reply:
Possibly, but I figured sooner or later we'd need this to be a Q_OBJECT and want to emit a signal

Reply:
I normally try to add overhead when it's needed and not prophylactic. Up to you.

Reply:
It's a singleton anyway. loosen up and have a :beer: !

Reply:
Alright :+1: 
I just had a couple of days off, back to :milk_glass: now :)